src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.cs

```diff
@@ -135,46 +129,25 @@ public int BinarySearch(T item, IComparer<T> comparer)
         /// cannot find an implementation of the <see cref="IComparable{T}"/> generic interface
         /// or the <see cref="IComparable"/> interface for type <typeparamref name="T"/>.
         /// </exception>
-        public int BinarySearch(int index, int count, T item, IComparer<T> comparer)
-        {
-            return _root.BinarySearch(index, count, item, comparer);
-        }
+        public int BinarySearch(int index, int count, T item, IComparer<T> comparer) => _root.BinarySearch(index, count, item, comparer);
 
         #region IImmutableList<T> Properties
 
         /// <summary>
         /// See the <see cref="IImmutableList{T}"/> interface.
         /// </summary>
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
-        public bool IsEmpty
-        {
-            get
-            {
-                Contract.Ensures(Contract.Result<bool>() == (this.Count == 0));
```

Comment:
@AArnott If you were referring to specifically Contracts and not assert validation in general, it seems like all Contract code is dead in corefx: https://github.com/dotnet/corefx/issues/11596